pkg/chartutil/coalesce.go

```diff
@@ -285,9 +286,49 @@ func coalesceTablesFullKey(printf printFn, dst, src map[string]interface{}, pref
 			} else {
 				printf("warning: cannot overwrite table with non table for %s (%v)", fullkey, val)
 			}
+		} else if islist(val) {
+			if islist(dv) {
+				dst[key] = coalesceListsFullKey(printf, dv.([]interface{}), val.([]interface{}), fullkey, merge)
+			} else {
+				printf("warning: cannot overwrite list with non list for %s (%v)", fullkey, val)
+			}
 		} else if istable(dv) && val != nil {
 			printf("warning: destination for %s is a table. Ignoring non-table value (%v)", fullkey, val)
 		}
 	}
 	return dst
 }
+
+// coalesceListsFullKey merges a source list into a destination list.
+//
+// dest is considered authoritative.
+func coalesceListsFullKey(printf printFn, dst, src []interface{}, prefix string, merge bool) []interface{} {
+	// When --reuse-values is set but there are no modifications yet, return new values
+	if src == nil {
+		return dst
+	}
+	if dst == nil {
+		return src
+	}
+	// Because dest has higher precedence than src, dest values override src
+	// values.
+	for key, val := range src {
+		if key == len(dst) {
```

Comment:
Just sanity checking my understanding this case is for when the destination list is smaller or empty compared to the source list?